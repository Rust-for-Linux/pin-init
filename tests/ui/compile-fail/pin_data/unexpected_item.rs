use pinned_init::*;

#[pin_data]
fn foo() {}

fn main() {}
