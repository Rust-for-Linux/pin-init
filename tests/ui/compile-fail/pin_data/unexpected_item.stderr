error: no rules expected the token `fn`
 --> tests/ui/compile-fail/pin_data/unexpected_item.rs:4:1
  |
4 | fn foo() {}
  | ^^ no rules expected this token in macro call
  |
note: while trying to match `struct`
 --> src/macros.rs
  |
  |             $vis:vis struct $name:ident
  |                      ^^^^^^

error: Could not locate type name.
 --> tests/ui/compile-fail/pin_data/unexpected_item.rs:3:1
  |
3 | #[pin_data]
  | ^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
